File: HelloWorld/task4/Task4Java/src/test/java/benchmark/SortsBenchmark.java
package benchmark;

import org.openjdk.jmh.annotations.*;

import java.util.Arrays;
import java.util.Random;
import java.util.concurrent.TimeUnit;
import Sorts.*;

import org.openjdk.jmh.annotations.*;
import org.openjdk.jmh.infra.Blackhole;

@State(Scope.Thread)
public class SortsBenchmark {

    @Param({"100", "10000"})
    private int N;
    private int[] array;

    @Setup(Level.Invocation)
    public void setUp() {
        array = new int[N];
        Random rand = new Random();
        for (int i = 0; i < array.length; i++) {
            array[i] = rand.nextInt();
        }
    }

    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.NANOSECONDS)
    @Fork(value = 1)
    @Measurement(iterations = 10)
    @Warmup(iterations = 1)
    public void QuickSortTest(){
        QuickSort.quickSort(array, 0, N-1);
    }

    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.NANOSECONDS)
    @Fork(value = 1)
    @Measurement(iterations = 10)
    @Warmup(iterations = 1)
    public void MergeSortTest(){
        MergeSort.mergeSort(array);
    }

    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.NANOSECONDS)
    @Fork(value = 1)
    @Measurement(iterations = 10)
    @Warmup(iterations = 1)
    public void BubbleSortTest(){
        BubbleSort.sort(array);
    }

    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.NANOSECONDS)
    @Fork(value = 1)
    @Measurement(iterations = 10)
    @Warmup(iterations = 1)
    public void StandartSortTest(Blackhole bh){ Arrays.sort(array); bh.consume(array); }
}

File: HelloWorld/task4/Task4Cs/BenchmarkTest/SortsBenchmark.cs
using System;
using BenchmarkDotNet.Attributes;
using Task4Cs;

namespace BenchmarkTest
{
    [MemoryDiagnoser]
    public class SortsBenchmark
    {
        [Params(100, 10000)] 
        public int N;
        public int[] array;

        [GlobalSetup]
        public void Setup()
        {
            array = new int[N];
            Random rand = new Random();
            for (int x = 0; x < array.Length; x++)
            {
                array[x] = rand.Next();
            }
        }

        [Benchmark]
        public void QuickSortTest()
        {
            QuickSort.Run(array);
        }
        
        [Benchmark]
        public void MergeSortTest()
        {
            MergeSort.Run(array);
        }
        
        [Benchmark]
        public void StandardSortTest()
        {
            Array.Sort(array);
        }
        
        [Benchmark]
        public void BubbleSortTest()
        {
            BubbleSort.Run(array);
        }
    }
}